roles/openshift_cluster_autoscaler/templates/cluster-autoscaler.j2

```diff
@@ -0,0 +1,75 @@
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    app: {{ openshift_cluster_autoscaler_name }}
+  name: {{ openshift_cluster_autoscaler_name }}
+  namespace: {{ openshift_cluster_autoscaler_namespace }}
+spec:
+  progressDeadlineSeconds: 600
+  replicas: {{ openshift_cluster_autoscaler_replicas }}
+  revisionHistoryLimit: 2
+  selector:
+    matchLabels:
+      app: {{ openshift_cluster_autoscaler_name }}
+      role: infra
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: {{ openshift_cluster_autoscaler_name }}
+        role: infra
+    spec:
+      containers:
+      - args:
+        - --alsologtostderr
+        - --v={{ openshift_cluster_autoscaler_version }}
+        - --skip-nodes-with-local-storage={{ openshift_cluster_autoscaler_skip_nodes_local_storage }}
+        - --leader-elect-resource-lock=configmaps
+        - --namespace={{ openshift_cluster_autoscaler_namespace }}
+{% if openshift_cluster_autoscaler_scale_down_delay is defined %}
+        - --scale-down-delay={{ openshift_cluster_autoscaler_scale_down_delay }}
+{% endif %}
+{% if openshift_cluster_autoscaler_cloud_provider == 'aws' %}
+        - --cloud-provider=aws
+{% endif %}
+{% for group in openshift_cluster_autoscaler_node_groups %}
+        - --nodes={{ group.min }}:{{ group.max }}:{{ group.name }}
+{% endfor %}
+{% if openshift_cluster_autoscaler_cloud_provider == 'aws' %}
```

Comment:
This seems clunky - have different env vars per provider

Reply:
Agreed, the credentials in AWS are required to be found in ~/.aws/credentials or in an environment variable.  Your example of gce didn't require credentials since your instance has a serviceaccount that allows the autoscaler access to make API calls.